### PR TITLE
Quick review

### PR DESCRIPTION
**Pull request by ldez**
I made a quick review:
- fix typo
- use embedded fields
- use golang-ci lint

Related to https://github.com/containous/traefik/pull/4755

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -25,16 +22,11 @@ type Conn struct {
 }
 
 // NewConn create wrapper on net.Conn.
-func NewConn(
-	conn net.Conn,
-	logger Logger,
-	headerParser HeaderParser,
-	trustedAddr bool,
-) net.Conn {
+func NewConn(conn net.Conn, logger Logger, headerParser HeaderParser, trustedAddr bool) net.Conn {
```
Comment (c0va23) [inline conn.go:25]: I don’t understand why multiline function declarations have been replaced with single-line ones. This makes reading difficult.

File: binary_receive.go [hunk 2]
```diff
@@ -37,20 +37,23 @@ func NewBinaryHeaderParser(logger Logger) BinaryHeaderParser {
...
 	}
 
-	if !bytes.Equal(magicBuf, BinarySignatue) {
+	if !bytes.Equal(magicBuf, BinarySignature) {
 		return nil, ErrInvalidSignature
 	}
 
-	buf.Discard(BinarySignatueLen)
+	_, err = buf.Discard(BinarySignatureLen)
+	if err != nil {
```
Comment (c0va23) [inline binary_receive.go:51]: Thanks for this fix.

**Review by c0va23 — CHANGES_REQUESTED** [review]
Thank you for correcting the typos. And embedding Conn simplifies the code.

But I don’t like the changes in the Makefile. I do not mind switching to golangci. But I do not like that the result of the Makefile depends on the golangci installed from outside.

File: conn.go (hunk 1 above)
Comment (ldez) [reply conn.go:25]: It depends on the writing habits: it is quite rare in Go to do this, but this is common in Ruby or JS.

**Comment by ldez** [comment]
Download binaries inside the project is not a standard way to use linters in Go, it's the first project where I see that way to manage linters.

In the Go eco-system use `go get` to install and use a binary is standard, then I don't think that use a tool from the "outside" of the project is a problem.

I will revert my changes but I don't think that's a good idea.

**Comment by c0va23** [comment]
@ldez, i am working in parallel with several Go projects that require different versions of linters and code generation utilities. With projects that locally install the necessary versions of utilities, it is much more pleasant to work.

**Review by c0va23 — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**